osquery/tables/system/linux/services.cpp

```diff
@@ -0,0 +1,585 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <algorithm>
+#include <cstdlib>
+#include <unordered_map>
+#include <vector>
+
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/path.hpp>
+#include <boost/process.hpp>
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+
+/*
+  You can compile this with the system libraries by uncommenting these lines
+  and removing "services.h"
+
+  #include <systemd/sd-bus.h>
+  #include <systemd/sd-daemon.h>
+  #include <systemd/sd-login.h>
+*/
+
+#include "services.h"
+using namespace systemd_deps;
+
+namespace osquery {
+namespace {
+
+namespace boostfs = boost::filesystem;
+namespace boostproc = boost::process;
+
+bool isSysVinitEnabled() {
+  auto enabled = boostfs::exists("/etc/init.d");
```

Comment:
Here's an example where you can use `osquery::pathExists`, or `osquery::isDirectory`.